examples/step-49/doc/results.dox

```diff
@@ -74,170 +74,152 @@ third direction, and gluing them together. The following code does this, using
 the techniques previously described:
 
 @code
-// Given a list of points and how vertices connect to cells,
-// create a mesh. This is in the same way as we do in step 14.
-void create_2d_grid (const Point<2> vertices_1[],
-	  	     const unsigned int n_vertices,
-		     const int cell_vertices[][4],
-		     const unsigned int n_cells,
-                     Triangulation<2> &coarse_grid)
+// Given a list of points and how vertices connect to cells, create a
+// mesh. This is in the same way as we do in step 14.
+void create_2d_grid(
+  const std::vector<Point<2>> &vertices,
+  const std::vector<
+    std::array<unsigned int, GeometryInfo<2>::vertices_per_cell>>
+    &               vertex_indices,
+  Triangulation<2> &coarse_grid)
 {
-  const std::vector<Point<2> > vertices (&vertices_1[0],
-			    	         &vertices_1[n_vertices]);
-
-  std::vector<CellData<2> > cells (n_cells, CellData<2>());
-  for (unsigned int i=0; i<n_cells; ++i)
+  std::vector<CellData<2>> cells(vertex_indices.size());
+  for (unsigned int i = 0; i < cells.size(); ++i)
     {
-      for (unsigned int j=0;
-	   j<GeometryInfo<2>::vertices_per_cell;
-	   ++j)
-	cells[i].vertices[j] = cell_vertices[i][j];
+      for (unsigned int j = 0; j < GeometryInfo<2>::vertices_per_cell; ++j)
+        cells[i].vertices[j] = vertex_indices[i][j];
     }
 
-  coarse_grid.create_triangulation (vertices,
-				    cells,
-				    SubCellData());
+  coarse_grid.create_triangulation(vertices, cells, SubCellData());
 }
 
 
 // Create a triangulation that covers the entire volume
-void create_3d_grid (Triangulation<3> &triangulation)
+void create_3d_grid(Triangulation<3> &triangulation)
 {
   // Generate first cross section
-  const Point<2> vertices_1[]
-    = {  Point<2> (-1.5,   0.),
-	 Point<2> (-0.5,   0.),
-	 Point<2> ( 0.5,   0.),
-	 Point<2> ( 1.5,   0.),
-
-	 Point<2> (-1.5,  1.5),
-	 Point<2> (-0.5,  1.5),
-	 Point<2> ( 0.5,  1.5),
-	 Point<2> ( 1.5,  1.5),
-
-	 Point<2> (-1.5,   3.),
-	 Point<2> (-0.5,   3.),
-	 Point<2> ( 0.5,   3.),
-	 Point<2> ( 1.5,   3.),
-
-	 Point<2> (-0.5,   3+0.5*sqrt(3)),
-	 Point<2> ( 0.5,   3+0.5*sqrt(3)),
-
-	 Point<2> (-0.75,  3+0.75*sqrt(3)),
-	 Point<2> ( 0.75,  3+0.75*sqrt(3))
-  };
-  const int cell_vertices_1[][GeometryInfo<2>::vertices_per_cell]
-    = {{0, 1, 4, 5},
-       {1, 2, 5, 6},
-       {3, 7, 2, 6},
-       {4, 5, 8, 9},
-       {5, 6, 9, 10},
-       {7,11, 6,10},
-       {8, 9, 14,12},
-       {9, 10,12,13},
-       {11,15,10,13},
-       {14,12,15,13}
-  };
+  std::vector<Point<2>> vertices_1{Point<2>(-1.5, 0.),
+                                   Point<2>(-0.5, 0.),
```

Comment:
I am surprised this actually works with just one set of braces. The braces here enclose constructor arguments, so I would have expected that you need a second set of braces to group all of these points to an initializer-list. Or that you write
```
  std::vector<...> vertices_1
  = { ...list of points ...};
```
We could also have made the things even simpler by replacing `Point<2>(-1.5,0)` to `{-1.5, 0}` and let the compiler do the conversion to `Point<2>` itself.

Reply:
I try not to ask too many questions about how the curly bracket initialization works.

Reply:
Yes, confusing indeed.